client/src/App/Header/MenuDesktop.tsx

```diff
@@ -30,7 +38,7 @@ const MenuDesktop: React.FC<MenuDesktopProps> = ({ loadingState, items }) => (
                 <LoadingBox state={loadingState} />
             </Grid>
             <Grid item>
-                <UserSelect />
+                <UserSelect currentUser={currentUser} onLogoutCb={onLogoutCb} />
```

Comment:
So here's an interesting thought... you're passing `onLogoutCb` few levels deep into the component tree. How about storing this in a Context?